### PR TITLE
chore: enforce import/no-extraneous-dependencies rule

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

Enforces and fixes violators of https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-extraneous-dependencies.md .

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sdstolworthy — DISMISSED** [review]

**Comment by sdstolworthy** [comment]
It's beautiful 🥲

**Review by goldbez — DISMISSED** [review]

File: packages/amplify-app/package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "@aws-amplify/amplify-frontend-flutter": "1.4.0",
     "@aws-amplify/amplify-frontend-ios": "3.6.1",
     "@aws-amplify/amplify-frontend-javascript": "3.8.1",
+    "@types/glob": "^7.1.1",
```
Comment (jhockett) [inline packages/amplify-app/package.json:35]: Can this be a devDependency?

File: packages/amplify-category-custom/package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "access": "public"
   },
   "dependencies": {
+    "@types/lodash": "^4.14.149",
```
Comment (jhockett) [inline packages/amplify-category-custom/package.json:29]: Can this be a devDependency?

File: packages/amplify-dotnet-function-template-provider/package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "extract-api": "ts-node ../../scripts/extract-api.ts"
   },
   "dependencies": {
+    "@types/lodash": "^4.14.149",
```
Comment (jhockett) [inline packages/amplify-dotnet-function-template-provider/package.json:28]: devDependency?

File: packages/amplify-go-function-template-provider/package.json [hunk 4]
```diff
@@ -25,6 +25,8 @@
     "extract-api": "ts-node ../../scripts/extract-api.ts"
   },
   "dependencies": {
+    "@types/fs-extra": "^8.0.1",
+    "@types/lodash": "^4.14.149",
```
Comment (jhockett) [inline packages/amplify-go-function-template-provider/package.json:29]: devDependency?

File: packages/amplify-nodejs-function-template-provider/package.json [hunk 5]
```diff
@@ -25,8 +25,10 @@
     "extract-api": "ts-node ../../scripts/extract-api.ts"
   },
   "dependencies": {
+    "@types/fs-extra": "^8.0.1",
```
Comment (jhockett) [inline packages/amplify-nodejs-function-template-provider/package.json:28]: devDependency?

File: packages/amplify-provider-awscloudformation/package.json [hunk 6]
```diff
@@ -35,6 +35,7 @@
     "@aws-amplify/cli-extensibility-helper": "3.0.0-beta.8",
     "@aws-amplify/graphql-transformer-core": "^1.1.0",
     "@aws-amplify/graphql-transformer-interfaces": "^2.1.0",
+    "@types/uuid": "^8.0.0",
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/package.json:38]: devDependency?

File: packages/amplify-python-function-runtime-provider/package.json [hunk 7]
```diff
@@ -25,6 +25,7 @@
     "extract-api": "ts-node ../../scripts/extract-api.ts"
   },
   "dependencies": {
+    "@types/fs-extra": "^8.0.1",
```
Comment (jhockett) [inline packages/amplify-python-function-runtime-provider/package.json:28]: devDependency?

File: packages/amplify-app/package.json (hunk 1 above)
Comment (sobolk) [reply packages/amplify-app/package.json:35]: it's main codebase
https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-app/src/xcodeHelpers.js#L2

File: packages/amplify-category-custom/package.json (hunk 2 above)
Comment (sobolk) [reply packages/amplify-category-custom/package.json:29]: it's here
https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-category-custom/src/utils/dependency-management-utils.ts#L6

File: packages/amplify-dotnet-function-template-provider/package.json (hunk 3 above)
Comment (sobolk) [reply packages/amplify-dotnet-function-template-provider/package.json:28]: no
https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-dotnet-function-template-provider/src/utils/destFileMapper.ts#L1

File: packages/amplify-go-function-template-provider/package.json (hunk 4 above)
Comment (sobolk) [reply packages/amplify-go-function-template-provider/package.json:29]: no 

https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-go-function-template-provider/src/utils/destFileMapper.ts#L1

https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-go-function-template-provider/src/providers/helloWorldProvider.ts#L3

File: packages/amplify-provider-awscloudformation/package.json (hunk 6 above)
Comment (sobolk) [reply packages/amplify-provider-awscloudformation/package.json:38]: https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-provider-awscloudformation/src/initializer.ts#L7

File: packages/amplify-python-function-runtime-provider/package.json (hunk 7 above)
Comment (sobolk) [reply packages/amplify-python-function-runtime-provider/package.json:28]: no
https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-python-function-runtime-provider/src/util/packageUtils.ts#L2

File: packages/amplify-python-function-template-provider/package.json [hunk 8]
```diff
@@ -25,6 +25,7 @@
     "extract-api": "ts-node ../../scripts/extract-api.ts"
   },
   "dependencies": {
+    "@types/fs-extra": "^8.0.1",
```
[the remark this answers is not in the document]
Comment (sobolk) [reply packages/amplify-python-function-template-provider/package.json:28]: https://github.com/aws-amplify/amplify-cli/blob/e5f00c392b150bbb47aed30c81cb5982b17c51a2/packages/amplify-python-function-template-provider/src/index.ts#L2

File: packages/amplify-app/package.json (hunk 1 above)
Comment (jhockett) [reply packages/amplify-app/package.json:35]: That's `glob` itself. The types are only consumed at build time, no?

**Comment by edwardfoyle** [comment]
John is right about all of the @types dev dependencies. Those are just typedefs that are only used at build time. The actual runtime packages are separate

**Review by edwardfoyle — CHANGES_REQUESTED** [review]
John is right about all of the `@types/*` dev dependencies. Those are just typedefs that are only used at build time. The actual runtime packages are separate

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-cli-core/package.json [hunk 9]
```diff
@@ -30,6 +30,7 @@
     "@yarnpkg/lockfile": "^1.1.0",
     "ajv": "^6.12.6",
     "amplify-prompts": "2.6.4",
+    "aws-cdk-lib": "~2.53.0",
```
Comment (sobolk) [inline packages/amplify-cli-core/package.json:33]: https://github.com/aws-amplify/amplify-cli/blob/e4cb68f2bde32459e4b33f833878a2c1c1b3088a/packages/amplify-cli-core/src/category-interfaces/amplify-base-cdk-types.ts#L2

**Review by sdstolworthy — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Pull request merged**